### PR TITLE
Add forceInit$ observable

### LINKED ISSUE #9: Editor fails to initialize if used inside of a component ng-content block that is hidden at first
### Issue description:
I've created a component that takes in content, an editable flag, and a read-only label:
`<app-ng-content-test [readOnlyContent]='readonly text' [editable]='isEditable'>
    <editor></editor>
</app-ng-content-test>`

Inside of this component (app-ng-content-test), I use an ngIf control block to show either my ng-content or my read only content. 
`<span *ngIf="editable; else readonlyBlock">
  <ng-content></ng-content>
</span>
<ng-template #readonlyBlock>
  {{readOnlyContent}}
</ng-template>`

When using a tinymce-angular editor inside of the component's content block, and defaulting the component to hide the editable content, the editor component fails to initialize, ever. When defaulting the component to show the editable content, the editor component initializes and works no matter how many times you toggle the wrapping component's editability. 

It seems like the editor component is not correctly detecting that it has been added to the DOM when used in this way.

Please see this branch in my fork for a working example: https://github.com/talikan/tinymce-angular/tree/ng-content-test

### PR DESCRIPTION
**Pull request by antoniogenaro**
I added an input variable in `editor.component` to force a reload of tinyEMC. This variable `forceInit$` is an observable that in its subscription calls a new remove method (`removeEditor()`) and `initialise()`.

`removeEditor` has the code that was inside ngOnDestroy method.

Also, I updated docs with the input variable and an example of how to use it.

I created this PR to fix #9. There is another PR to fix the same issue, #50, but is actually stopped and I don't think that add a timeout is a good solution. So my fix is only applied if you want use it manually.

Here is an example of how to use this feature.

In your template
```html
<editor [forceInit$]="forceInit$"></editor>
```

And your controller
```tsx
this.forceInit$ = new Subject();
this.forceInit$.next();
```

### CODE REVIEW

**Comment by SimonFc** [comment]
So the issue stems from the fact that Angular components can be created in a detached state where they're not contained in a document, right? We have gotten a few issues reported as you've already discovered so I would very much like to solve this although it feels like it's near impossible to create a simple solution.

The use case for your solution would be that you manually initialize Tiny when you, as the user of tinymce-angular, have decided that the component is attached to the document. Do I understand it correctly? It would still leave the issue of making sure that the parent component is attached as well so it's kind of delegating the issue, but perhaps this is overthinking it.

The best way would be to automate it if Angular were to provide ngOnAttach/ngOnDetach lifecycle hooks or something similar but unfortunately that doesn't seem to be the case. It really feels like something the users of tinymce-angular shouldn't have to take care of manually :thinking: 

Are you aware of any way we could make this process automatic and contained to tinymce-angular?

**Comment by antoniogenaro** [comment]
This isn't a perfect solution, because as you said, is not an automatic process. But at least, is a solution (the only one for now). You are right, when Angular detached the editor component, and is attached later (with `ngIf` in my case), the tinyMCE reference is lost. I think that when it happens, you have control of this event, at least in my case, all the times that it happens, is for a conditional (on `ngIf`) that I have the control of it. So, if you know why the component is detached, you can force a reload of tinyMCE.

In summary, this feature isn't the perfect solution, but you provide to the users a simple way to solve this problem.

However, in my last commit, I updated `forceInit` method to public, so you have a new way to force the reload. Here is a simple example.

```tsx
import { EditorComponent } from '@tinymce/tinymce-angular';

@ViewChild(EditorComponent) editor: EditorComponent;

this.editor.forceInit();
```

Thanks for your time and your answer!

**Comment by SimonFc** [comment]
@antoniogenaro I'm looking into what alternative solutions there might be, do you have an example of your setup where you encountered this issue? Either on [codesandbox](https://codesandbox.io/s/6zlmmkkk1n?fontsize=14) or as a short example here. If it's the same as #9 then no reproduction is needed - just let me know. Thanks!

**Comment by SimonFc** [comment]
@antoniogenaro So I looked into possible workarounds to the problem and it seems like there are one. I made a scaled-down version of the example provided in PR #9 - [codesandbox](https://codesandbox.io/embed/tinymceangular-tn812?fontsize=14). Instead of using ng-content to achieve transclusion I used [NgTemplateOutlet](https://angular.io/api/common/NgTemplateOutlet) and the behavior seems to be the sought after one.

It doesn't seem to be possible to conditionally use ng-content as shown in the other PR due to angular always instantiating the components whether or not they're attached to the DOM. At least until the day Angular decides to introduce ngOnAttachDom/ngOnDetachDom hooks. It would be possible to make something similar to that using Node.contains/Node.isConnected today though and guard against it ourselves in tinymce-angular but I fear that it could be computationally expensive.

And if there are work arrounds to this issue I would rather not introduce a new API. Would the example I made solve your use case?

**Comment by techtangents** [comment]
Hi,

Due to the age of this PR, we're going to close it. If you continue to experience issues, please review the previous suggestion and feel free to raise another issue describing remaining issues.

Kind regards,
Dylan Just
Tiny
